pkg/credentialprovider/config.go

```diff
@@ -73,11 +73,23 @@ func GetPreferredDockercfgPath() string {
 	defer preferredPathLock.Unlock()
 	return preferredPath
 }
+func GetDockercfgJsonPaths() []string {
+	return []string{GetPreferredDockercfgPath(), workingDirPath, homeJsonDirPath, rootJsonDirPath}
+}
+func GetDockercfgFilePaths() []string {
+	return []string{GetPreferredDockercfgPath(), workingDirPath, homeDirPath, rootDirPath}
+}
+
+func ReadDockerConfigFile(paths []string) (cfg DockerConfig, err error) {
```

Comment:
@liggitt yeah, through it will find some no relative paths when using a single intersect path set, it looks like a good way.